app/views/common/_login.html.erb

```diff
@@ -0,0 +1,2 @@
+<div id='login-page'></div>
+<%= mount_react_component('ComponentWrapper', '.login-page', { component: 'LoginPage', componentProps: props }.to_json)%>
```

Comment:
Does it have to use the `ComponentWrapper`? Can't you pass those props as `data`?
Then we will know which props came from the server and which came from the client.

imo the only reason to use the `ComponentWrapper`  is when you want to be able to set the same props from the server and from the client.

Reply:
IMO without the data object, prop types are much simpler to maintain, you need less component containers, it is easier to consume the component.. we can manage the props from the server by `componentDidMount` or by `selectors` through `mapStateToProps`

Reply:
I don't mind sticking for our patterns,
but I really think that it makes things more complex..
for example the background for the login page: 
the default prop is the background that you see here,
from the server we get nothing in this case, and we've moved it to props so it would be easier for the branding team.

in case I will use the data object, the background from the server would be undefined but it is now nested under the data object, and it isn't required. so my option is to create a container and pass the data object destructed, or to start and test if I received the background from the server else use some default, or to declare duplicated props for in/out the data object.